app/models/subscription.rb

```diff
@@ -623,7 +623,12 @@ def formatted_end_time_of_subscription
 
   def recurrence
     if is_installment_plan
-      last_payment_option.installment_plan.recurrence
+      # Use snapshot to preserve original schedule frequency, fallback to live plan if no snapshot exists
```

Comment:
This methods becomes hard to read. Refactor it using guard clauses (make sure the following suggestion works):
```ruby
def recurrence
  return price.recurrence unless is_installment_plan
  return last_payment_option.installment_plan.recurrence unless last_payment_option&.installment_plan_snapshot

  last_payment_option.installment_plan_snapshot.recurrence
end